chia/rpc/util.py

```diff
@@ -2,18 +2,41 @@
 
 import logging
 import traceback
-from typing import Any, Callable, Coroutine, Dict, List, Optional, Tuple
+from typing import Any, Awaitable, Callable, Dict, List, Optional, Tuple, Type, get_type_hints
 
 import aiohttp
 
 from chia.types.blockchain_format.coin import Coin
 from chia.util.json_util import obj_to_response
+from chia.util.streamable import Streamable
 from chia.wallet.conditions import Condition, ConditionValidTimes, conditions_from_json_dicts, parse_timelock_info
 from chia.wallet.transaction_record import TransactionRecord
 from chia.wallet.util.tx_config import TXConfig, TXConfigLoader
 
 log = logging.getLogger(__name__)
 
+# TODO: consolidate this with chia.rpc.rpc_server.Endpoint
+# Not all endpoints only take a dictionary so that definition is imperfect
+# This definition is weaker than that one however because the arguments can be anything
+RpcEndpoint = Callable[..., Awaitable[Dict[str, Any]]]
+MarshallableRpcEndpoint = Callable[..., Awaitable[Streamable]]
+
+
+def marshal(func: MarshallableRpcEndpoint) -> RpcEndpoint:
+    hints = get_type_hints(func)
+    request_class: Type[Streamable] = hints["request"]
```

Comment:
Might be worth checking here that this is true?  Then we would get an import time exception for mistakes.

```suggestion
    request_hint = hints["request"]
    assert isinstance(request_hint, Streamable)
    request_class = request_hint
```

We _might_ also be able to get to a way to hint this requirement.  Maybe.